great_expectations/data_context/store/_store_backend.py

```diff
@@ -0,0 +1,212 @@
+            suppress_store_backend_id: skip construction of a StoreBackend.store_backend_id
+            manually_initialize_store_backend_id: UUID as a string to use if the store_backend_id is not already set
+            store_name: store name given in the DataContextConfig (via either in-code or yaml configuration)
+        """
+        self._fixed_length_key = fixed_length_key
+        self._suppress_store_backend_id = suppress_store_backend_id
+        self._manually_initialize_store_backend_id = (
+            manually_initialize_store_backend_id
+        )
+        self._store_name = store_name
+
+    @property
+    def fixed_length_key(self):
+        return self._fixed_length_key
+
+    @property
+    def store_name(self):
+        return self._store_name
+
+    def _construct_store_backend_id(
+        self, suppress_warning: bool = False
+    ) -> Optional[str]:
+        """
+        Create a store_backend_id if one does not exist, and return it if it exists
+        If a valid UUID store_backend_id is passed in param manually_initialize_store_backend_id
+        and there is not already an existing store_backend_id then the store_backend_id
+        from param manually_initialize_store_backend_id is used to create it.
+        Args:
+            suppress_warning: boolean flag for whether warnings are logged
+
+        Returns:
+            store_backend_id which is a UUID(version=4)
+        """
+        if self._suppress_store_backend_id:
+            if not suppress_warning:
+                logger.warning(
+                    f"You are attempting to access the store_backend_id of a store or store_backend named {self.store_name} that has been explicitly suppressed."
+                )
+            return None
+        try:
+            try:
+                ge_store_backend_id_file_contents = self.get(
+                    key=self.STORE_BACKEND_ID_KEY
+                )
+                store_backend_id_file_parser = self.STORE_BACKEND_ID_PREFIX + pp.Word(
+                    f"{pp.hexnums}-"
+                )
+                parsed_store_backend_id = store_backend_id_file_parser.parseString(
+                    ge_store_backend_id_file_contents
+                )
+                return parsed_store_backend_id[1]
+            except InvalidKeyError:
+                store_id = (
+                    self._manually_initialize_store_backend_id
+                    if self._manually_initialize_store_backend_id
+                    else str(uuid.uuid4())
+                )
+                self.set(
+                    key=self.STORE_BACKEND_ID_KEY,
+                    value=f"{self.STORE_BACKEND_ID_PREFIX}{store_id}\n",
+                )
+                return store_id
+        except Exception:
+            if not suppress_warning:
+                logger.warning(
+                    f"Invalid store configuration: Please check the configuration of your {self.__class__.__name__} named {self.store_name}"
```

Comment:
This seems serious, should we die? Basically, I'm wary of catching exceptions, logging and moving on. Does the process end up in a bad state or is everything ok?